pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -132,3 +133,24 @@ type PipelineList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Pipeline `json:"items"`
 }
+
+// GetTask is a function that will retrieve the Task name from namespace.
+type GetTask func(namespace, name string) (*Task, error)
```

Comment:
why do we need the func as a type here?

Reply:
I guess the alternative would be to declare it in the function signature directly, like this:

```go
func (p *Pipeline)GetTasks(g func(namespace, name string) (*Task, error)) (map[string]*Task, error) {
```

? 

Or are you asking why we are depending on a function at all? (Doing this so we can swap out the implementation easily in tests and also so this package doesn't need to know about Listers in general)